src/ProjectTemplates/Shared/Project.cs

```diff
@@ -102,6 +102,21 @@ public class Project : IDisposable
             try
             {
                 Output.WriteLine("Acquired DotNetNewLock");
+                
+                if (Directory.Exists(TemplateOutputDir))
```

Comment:
I can file a separate issue to track changing that if you want, but I don't really want to do additional major surgery (prefer making this PR just about adding/using retries)

Reply:
There's additional complications about switching from reusing projects since our longer term hope was to actually switch to one test creating/building the template project, and a second (ordered) test would be in charge of running/verifying the tests, we were hoping that would also reduce the flakiness, or at least help us isolate which part of the test is flaky.  Unfortunately we didn't get to that point yet, but that was the longer term goal